src/CoreBundle/Bridge/Chameleon/Migration/Script/update-1549367836.inc.php

```diff
@@ -0,0 +1,319 @@
+      'fieldtype_config' => '',
+      'restrict_to_groups' => '0',
+      'field_width' => '',
+      'position' => '0',
+      '049_helptext' => '',
+      'row_hexcolor' => '',
+      'is_translatable' => '0',
+      'validation_regex' => '',
+      'id' => $nameFieldId,
+  ])
+;
+TCMSLogChange::insert(__LINE__, $data);
+
+$query = 'ALTER TABLE `cms_menu_item`
+                        ADD `name` VARCHAR(255) NOT NULL';
+TCMSLogChange::RunQuery(__LINE__, $query);
+
+TCMSLogChange::makeFieldMultilingual('cms_menu_item', 'name');
+
+$data = TCMSLogChange::createMigrationQueryData('cms_field_conf', 'de')
+  ->setFields([
+      'translation' => 'Name',
+  ])
+  ->setWhereEquals([
+      'id' => $nameFieldId,
+  ])
+;
+TCMSLogChange::update(__LINE__, $data);
+
+// Add target field.
+
+$targetFieldId = TCMSLogChange::createUnusedRecordId('cms_field_conf');
+
+$data = TCMSLogChange::createMigrationQueryData('cms_field_conf', 'en')
+  ->setFields([
+      'cms_tbl_conf_id' => $cmsMenuItemTableId,
+      'name' => 'target',
+      'translation' => '',
+      'cms_field_type_id' => TCMSLogChange::GetFieldType('CMSFIELD_EXTENDEDMULTITABLELIST'),
+      'cms_tbl_field_tab' => '',
+      'isrequired' => '0',
+      'fieldclass' => '',
+      'fieldclass_subtype' => '',
+      'class_type' => 'Core',
+      'modifier' => 'none',
+      'field_default_value' => '',
+      'length_set' => '',
+      'fieldtype_config' => 'sTables=cms_tbl_conf,cms_module,cms_menu_custom_item',
+      'restrict_to_groups' => '0',
+      'field_width' => '',
+      'position' => '0',
+      '049_helptext' => '',
+      'row_hexcolor' => '',
+      'is_translatable' => '0',
+      'validation_regex' => '',
+      'id' => $targetFieldId,
+  ])
+;
+TCMSLogChange::insert(__LINE__, $data);
+
+$query = 'ALTER TABLE `cms_menu_item`
+                        ADD `target` CHAR(36) CHARACTER SET latin1 COLLATE latin1_general_ci NOT NULL';
+TCMSLogChange::RunQuery(__LINE__, $query);
+
+$query = 'ALTER TABLE `cms_menu_item`
+                         ADD `target_table_name` VARCHAR(64) CHARACTER SET latin1 COLLATE latin1_general_ci NOT NULL';
+TCMSLogChange::RunQuery(__LINE__, $query);
+
+$data = TCMSLogChange::createMigrationQueryData('cms_field_conf', 'de')
+  ->setFields([
+      'translation' => 'Ziel',
+  ])
+  ->setWhereEquals([
+      'id' => $targetFieldId,
+  ])
+;
+TCMSLogChange::update(__LINE__, $data);
+
+$query = 'ALTER TABLE `cms_menu_item` ADD INDEX ( `target` )';
+TCMSLogChange::RunQuery(__LINE__, $query);
+
+$query = 'ALTER TABLE  `cms_menu_item` ADD INDEX `target_combined` (  `target_table_name` ,  `target` )';
+TCMSLogChange::RunQuery(__LINE__, $query);
+
+// Add icon_font_css_class field.
+
+$iconFontCssClassFieldId = TCMSLogChange::createUnusedRecordId('cms_field_conf');
+
+$data = TCMSLogChange::createMigrationQueryData('cms_field_conf', 'en')
+  ->setFields([
+      'cms_tbl_conf_id' => $cmsMenuItemTableId,
+      'name' => 'icon_font_css_class',
```

Comment:
Is this a css class specifiying the icon (ie a symbol of a font).

Or is it an additional class further specifying the icon  - that's what the name sounds like?

Reply:
It's one or more CSS classes that depict the icon from the system's icon font to be used for this menu item, e.g. "fas fa-leaf" for a leaf icon in FontAwesome.